editor/create_dialog.cpp

```diff
@@ -330,7 +330,7 @@ void CreateDialog::_update_search() {
 					break;
 				}
 
-				type = ClassDB::get_parent_class(type);
+				type = (cpp_type ? ClassDB::get_parent_class(type) : "");
```

Comment:
I'm not convinced this is the correct fix, it defeats the purpose of the while loop so if `""` is what is expected for a non-C++ class, some refactoring would be in order.

Reply:
Previously it did indeed return "" for non C++ classes

Reply:
@DualMatrix @akien-mga The proper fix for this is to handle the script-supported "get_parent_class" variant present in EditorData, just like is done in the while loop statement (apologies, I must've missed this during my testing). Rather than hardcoding an empty-string response, it should be `ed.script_class_get_base(type)` since this method will check if the script class has an inherited script and whether that inherited script also is a script class to be added to the tree.

Reply:
The ScriptServer method will only ever return the base C++ type (since that's all that's needed to create the object). For ClassDB-simulated methods, you have to use the EditorData instance.